SCClassLibrary/Common/Quarks/Quark.sc

```diff
@@ -78,6 +78,13 @@ Quark {
 		});
 		^isCompatible
 	}
+	runHook { |hook|
+		hook = hook.asSymbol;
+		if(this.data[hook].notNil, {
+			"Run % hook".format(hook).postln;
+			this.data[hook].();
+		});
```

Comment:
Pass `data` in as the first argument (data is the Quark data structure here, right?). Without this, it's a little hard for a hook to access that information.